extensions/oplogPopulator/KafkaConnectWrapper.js

```diff
@@ -0,0 +1,322 @@
+const { errors } = require('arsenal');
+const async = require('async');
+const joi = require('joi');
+const http = require('http');
+
+const paramsJoi = joi.object({
+    kafkaConnectHost: joi.string().required(),
+    kafkaConnectPort: joi.number().required(),
+    logger: joi.object().required(),
+}).required();
+
+/**
+ * @class KafkaConnectWrapper
+ *
+ * @classdesc Wrapper arround the Kafka Connect REST API.
+ */
+class KafkaConnectWrapper {
+
+    /**
+     * @constructor
+     * @param {Object} params constructor params
+     * @param {string} params.kafkaConnectHost kafka connect host
+     * @param {number} params.kafkaConnectPort kafka connect port
+     * @param {Logger} params.logger - logger object
+     */
+    constructor(params) {
+        joi.attempt(params, paramsJoi);
+        this._kafkaConnectHost = params.kafkaConnectHost;
+        this._kafkaConnectPort = params.kafkaConnectPort;
+        this._logger = params.logger;
+    }
+
+    /**
+     * makes http request
+     * @param {Object} params request params
+     * @param {string} params.host host name
+     * @param {string} params.port port
+     * @param {string} params.method http method
+     * @param {string} params.path path to resource
+     * @param {string} params.headers request headers
+     * @param {Object} data request data
+     * @returns {Promise} http response
+     */
+    makeRequest(params, data) {
+        return new Promise((resolve, reject) => {
```

Comment:
Late to the party, but the following describes the idea: https://scality.atlassian.net/wiki/spaces/OS/pages/1869774969/TypeScript+package+maintenance+strategies#The-use-of-%40deprecated

Essentially, this means exposing 2 overloads of the same function - one returning a promise, the other a callback - and making TS connect the dots, with a deprecation on the function call we want to remove.

This doesn't work in pure JS, because JS doesn't have overloading.